### PR TITLE
Settings tests

### PR DESCRIPTION
**Pull request by Warhedgehog**
This adds test for the settings screen namely that the components are displayed, that when modifying the server URL the apply button is enabled and clicking on it goes to Home

### CODE REVIEW

**Review by Tesa27 — DISMISSED** [review]
LGTM

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
Looks good !

I just have a small proposal regarding test rule syntax.

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/ui/settings/SettingsTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@HiltAndroidTest
+@RunWith(AndroidJUnit4.class)
+public class SettingsTest {
+
+  private final FragmentScenarioRule<SettingsFragment> fragmentRule =
+      FragmentScenarioRule.launch(SettingsFragment.class);
+
+  private final HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+  @Rule public final RuleChain chain = RuleChain.outerRule(hiltRule).around(fragmentRule);
```
[the remark this answers is not in the document]
Comment (Warhedgehog) [reply fe2-android/app/src/test/java/com/github/dedis/popstellar/ui/settings/SettingsTest.java:37]: Good suggestion thanks

**Review by GabrielFleischer — APPROVED** [review]
LGTM

**Pull request merged**